### ISSUE TITLE
bug: `npx cap run iOS` does not deploy app to physical device

### ISSUE THREAD

**Issue by qpre** [ionic-team/capacitor#6254]
### Capacitor Version

```
npx cap doctor ios
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 4.6.2
  @capacitor/core: 4.6.2
  @capacitor/android: 4.6.2
  @capacitor/ios: 4.6.2

Installed Dependencies:

  @capacitor/cli: 4.6.2
  @capacitor/core: 4.6.2
  @capacitor/ios: 4.6.2
  @capacitor/android: 4.6.2

[success] iOS looking great! 👌
```

### Platform(s)
- iOS

### Current Behavior

When `npx cap run iOS` targets a simulator, everything works as expected.

When `npx cap run iOS` targets a physical device, the command gives a success output (see below), but the app is not deployed on the device.

```
✔ Copying web assets from out to ios/App/App/public in 98.45ms
✔ Creating capacitor.config.json in ios/App/App in 807.00μs
[info] Found 1 Cordova plugin for ios:
       pushy-cordova@1.0.51
[info] Inlining sourcemaps
✔ copy ios in 166.14ms
✔ Updating iOS plugins in 4.70ms
[info] Found 14 Capacitor plugins for ios:
       @byteowls/capacitor-filesharer@4.0.0
       @capacitor-community/bluetooth-le@2.1.0
       @capacitor-community/keep-awake@3.0.0
       @capacitor/app@4.1.1
       @capacitor/camera@4.1.4
       @capacitor/device@4.1.0
       @capacitor/dialog@4.1.0
       @capacitor/keyboard@4.1.1
       @capacitor/local-notifications@4.1.4
       @capacitor/network@4.1.0
       @capacitor/preferences@4.0.2
       @capacitor/share@4.1.0
       @capacitor/status-bar@4.1.1
       @sentry/capacitor@0.11.0
[info] Found 1 Cordova plugin for ios:
       pushy-cordova@1.0.51
✔ Updating iOS native dependencies with pod install in 2.24s
```

Is there a way to run the command in a verbose mode ?

My setup uses XCode Version 14.2 (14C18), might it be linked ? (though it works fine when using the "run" button in XCode) 
 
### Expected Behavior

The app should be deployed to the phone.

### Code Reproduction

I get the exact same result with `@capacitor/create-app`;

### Other Technical Details

`npm --version` output: 8.19.2

`node --version` output: v18.12.1

`pod --version` output (iOS issues only): 1.11.3 (homebrew on M1 Mac)

device iOS version: 16.3

**Comment by rgustavsson** [comment]
Seeing the same problem. Would love to hear about workarounds!

**Comment by rgustavsson** [comment]
@qpre did you find a workaround?
I downgraded from node 19 to 16 and then it started to work for us.

**Comment by qpre** [comment]
Hey, I did not find a workaround, I currently prepare the web app then deploy it with Xcode, which is cumbersome.


I'm using node `v18.14.0` which is the latest LTS from node.

**Comment by rgustavsson** [comment]
Yeah ok.
Downgrade node and the terminal will probably work for you. I know v18 is
latest stable, but that has nothing with capacitor to do.

I describe my issue here:
https://www.appsari.com/ionic-issues-solved/cant-deploy-ios-device

lör 11 feb. 2023 kl. 09:09 skrev Quentin Pré ***@***.***>:

**Comment by qpre** [comment]
I'd rather click a button until then than downgrade nodejs :)

**Comment by rgustavsson** [comment]
I use ionic with live reload and so terminal is a must, unless your fine
with building and deploying the app with every change you want to test.

fre 17 feb. 2023 kl. 08:10 skrev Quentin Pré ***@***.***>:

**Comment by qpre** [comment]
I do not, `npx cap sync iOS` works, I launch my web server then sync then build with Xcode only once. 

I only have to rebuild with Xcode if I add a native plugin.

**Comment by markemer** [comment]
If you use the latest `native-run` 1.7.2, it should work now. `package.json` suggests `^1.6.0` so you may be able to get away with an `npm install`

**Comment by qpre** [comment]
hey @markemer thanks for the update, pinning `native-run` to 1.7.2 brings the process a bit further as it fails but not silently anymore.

```
[error] Error: There was an error installing app
        at
        ***/node_modules/native-run/dist/ios/lib/client/installation_proxy.js:72:24
        at LockdownProtocolReader.callback
        (***/node_modules/native-run/dist/ios/lib/protocol/protocol.js:93:21)
        at LockdownProtocolReader.onData
        (***/node_modules/native-run/dist/ios/lib/protocol/protocol.js:53:26)
        at Socket.emit (node:events:513:28)
        at addChunk (node:internal/streams/readable:324:12)
        at readableAddChunk (node:internal/streams/readable:297:9)
        at Readable.push (node:internal/streams/readable:234:10)
        at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
```

The app appears on the device but is not launchable.

**Comment by markemer** [comment]
Well, it appears you had two problems - this one is a bit trickier. The fact that it installs but doesn't run sounds like a provisioning profile thing, but it does run from inside Xcode?

**Comment by markemer** [comment]
Also, could you try a fresh project and see if that also has the same problem? Or post a small sample that is failing for you?

**Comment by qpre** [comment]
It actually works from a freshly created capacitor app 🤔

Do you know anyway I could get a more verbose output from the run command (like seeing the original trace that lead to "Error: There was an error installing app") ?

**Comment by markemer** [comment]
> It actually works from a freshly created capacitor app 🤔
> 
> 
> 
> Do you know anyway I could get a more verbose output from the run command (like seeing the original trace that lead to "Error: There was an error installing app") ?

Ok that's good news and bad news. So it's something special about your project. `native-run` accepts a verbose option, but I think you have to call it directly. 

It's on my todo list to bubble that functionality up to `npx cap` but that doesn't help us today. Is there a subset of a project that doesn't work you can share that I can take a look at?

**Comment by ToschaA** [comment]
Same problem here!

**Comment by markemer** [comment]
> Same problem here! EDIT: downgrading to node@16 worked for me as well!

@ToschaA The new error, that @qpre is seeing? or the old one?

**Comment by ToschaA** [comment]
> > Same problem here! EDIT: downgrading to node@16 worked for me as well!
> 
> @ToschaA The new error, that @qpre is seeing? or the old one?

The old one (first)!

**Comment by markemer** [comment]
> > > Same problem here! EDIT: downgrading to node@16 worked for me as well!
> > 
> > 
> > @ToschaA The new error, that @qpre is seeing? or the old one?
> 
> The old one (first)!

Making sure native-run is at 1.7.2 should fix things - this is enforced in Capacitor 4.7.1 as well.

**Comment by markemer** [comment]
@qpre can you open a new ticket with this second problem to better keep track? I'm going to close this as we fixed the original issue.
